packages/server/src/requestPipeline.ts

```diff
@@ -297,11 +297,13 @@ export async function processGraphQLRequest<TContext extends BaseContext>(
     // While it shouldn't normally be necessary to wrap this `Promise` in a
     // `Promise.resolve` invocation, it seems that the underlying cache store
     // is returning a non-native `Promise` (e.g. Bluebird, etc.).
+    const ttl = internals.persistedQueries?.ttl;
     Promise.resolve(
       internals.persistedQueries.cache.set(
         queryHash,
         query,
-        internals.persistedQueries?.ttl ?? undefined,
+        // careful with zero here, explicitly checking for `null` or `undefined`
```

Comment:
hmmmmm.

so looking at the AS3 code, if persistedQueries.ttl is null (not undefined) it will actually pass it in as `{ttl: null}`.

And... In the redis implementation, we have a hardcoded default TTL of 300 (why???) which is overridden by {ttl: null} but not if you don't pass ttl at all.

tbh this is all kinda weird — it would make a lot more sense if `{ttl: null}` was equivalent to not specifying it. and I don't know why the redis adapter has a default TTL at all!

so I think this change from AS3 is reasonable — just something we should probably note in migration or something.

Reply:
actually let's split this into two separate concepts.

(a) Super weird that Redis has a default TTL of 300. When you switch to the keyv version you won't have that; let's make sure migration guide points that out and lets you set it yourself.

(b) That said, it is reasonable for caches to have a user-configurable default TTL, and Keyv supports that.  So it is not unreasonable to be in a situation where you set up your cache with default TTL 300 (for use with data sources, full response cache, etc) but you actually want unlimited TTL for persisted queries (because it's a hash inversion).  So supporting `persistedQueries: {ttl: null}` does seem like an actual feature to be careful to continue to support.